modules/Product/app/Services/ProductService.php

```diff
@@ -63,7 +68,26 @@ public function show(Product $product)
 
     public function update(Request $request, Product $product)
     {
-        return $this->productRepository->update($request->all(), $product);
+        $userId = auth()->id();
+        $requestData = $request->all();
+
+        $data = [
+            'name' => $requestData['name'],
+            'description' => $requestData['description'],
+            'price' => $requestData['price'],
+            'stock' => $requestData['stock'],
+            'status' => $requestData['status'],
+            'updated_by' => $userId,
+            'categories' => $requestData['categories']
+        ];
+
+        $product = $this->productRepository->update($data, $product);
+
+        $product->load('categories', 'files');
```

Comment:
You have this line in the create method in the repository, but here, you use it in the service. Please use one approach. I think it is better to use it in the repository.